src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java

```diff
@@ -201,23 +213,31 @@ private Object traverseField(String fieldName, Value fieldValue) {
                 .collect(Collectors.toList());
     }
 
-    private void traverseField(IMessage message, String fieldName, Value value, IFieldStructure fieldStructure) {
-        if (fieldStructure != null) {
-            Object convertedValue = fieldStructure.isComplex()
-                    ? processComplex(value, fieldStructure)
-                    : convertSimple(value, fieldStructure);
-            message.addField(fieldName, convertedValue);
-        }
+    private void traverseField(IMessage message, String fieldName, Value value, @NotNull IFieldStructure fieldStructure) {
+        Object convertedValue = fieldStructure.isComplex()
+                ? processComplex(value, fieldStructure)
+                : convertSimple(value, fieldStructure);
+        message.addField(fieldName, convertedValue);
     }
 
-    private Object convertSimple(Value value, IFieldStructure fieldStructure) {
+    private Object convertSimple(Value listEvent, IFieldStructure fieldStructure) {
         if (fieldStructure.isCollection()) {
-            return value.getListValue().getValuesList()
-                    .stream()
-                    .map(element -> convertToTarget(element, fieldStructure))
-                    .collect(Collectors.toList());
+            checkKind(listEvent, fieldStructure.getName(), KindCase.LIST_VALUE);
+
+            List<Value> valuesList = listEvent.getListValue().getValuesList();
+            int size = valuesList.size();
+            List<Object> result = new ArrayList<>(size);
+            for (int i = 0; i < size; i++) {
+                Value element = valuesList.get(i);
+                try {
+                    result.add(convertToTarget(element, fieldStructure));
```

Comment:
Maybe we need to check that the kind is SIMPLE_VALUE as well

Reply:
The method convertToTarget contains several checks related to value